decidim-core/lib/decidim/common_passwords.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Decidim
+  class CommonPasswords
+    include Singleton
+
+    attr_reader :passwords
+
+    URLS = %w(
+      https://raw.githubusercontent.com/danielmiessler/SecLists/master/Passwords/xato-net-10-million-passwords-1000000.txt
+      https://raw.githubusercontent.com/danielmiessler/SecLists/master/Passwords/darkweb2017-top10000.txt
+      https://raw.githubusercontent.com/danielmiessler/SecLists/master/Passwords/Common-Credentials/10-million-password-list-top-1000000.txt
+    ).freeze
+
+    def initialize
+      update_passwords! unless File.exist?(password_list_path)
+
+      File.open(password_list_path, "r") do |file|
+        @passwords = file.read.split
```

Comment:
The password list file can contain the final empty line, so let's filter empty lines away from the list.